### PR TITLE
feat(Contacts): select billing contact for sales invoice

### PR DESCRIPTION
**Pull request by ronelvcabrera**
Reopening PR -  https://github.com/frappe/erpnext/pull/20210

Most of the time, point of contact are different from sale, invoicing and on different aspect/phase of the transaction. This feature is to have contacts that are identified for billing.

Therefore, as seen below, when creating a sales invoice, the priority to retrieve a contact for Sales invoice are tagged as `is Billing Contact` contacts.

In connection with this PR https://github.com/frappe/frappe/pull/8840

### Usage:

Billing contact will be retrieved during creation of Invoice as point of contact at this stage is different from sale or other aspect/stage of the transaction.

core is billing contact new contact

### CODE REVIEW

**Comment by ronelvcabrera** [comment]
Working on Build failure

**Comment by ronelvcabrera** [comment]
set working copy to `develop` branch and ran `...run-tests...` seems failures exists already without my changes.

File: erpnext/erpnext_integrations/custom/contact.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+   "columns": 0,
+   "creation": "2019-12-02 11:00:03.432994",
+   "default": null,
+   "depends_on": null,
+   "description": null,
+   "docstatus": 0,
+   "dt": "Contact",
+   "fetch_from": null,
+   "fetch_if_empty": 0,
+   "fieldname": "is_billing_contact",
```
Comment (deepeshgarg007) [inline erpnext/erpnext_integrations/custom/contact.json:21]: @ronelvcabrera why are you adding `is_billing_contact` as a custom field and exporting as fixtures and why not as a standard field in contact doctype

Comment (ronelvcabrera) [reply erpnext/erpnext_integrations/custom/contact.json:21]: @deepeshgarg007 As advised on this PR comment https://github.com/frappe/frappe/pull/8840#issuecomment-558469686

File: erpnext/accounts/party.py [hunk 2]
```diff
@@ -613,3 +616,26 @@ def get_partywise_advanced_payment_amount(party_type, posting_date = None):
...
+	out = frappe.db.sql("""
+			SELECT dl.parent, c.is_primary_contact, c.is_billing_contact
+			FROM `tabDynamic Link` dl
+			INNER JOIN tabContact c ON c.name = dl.parent
+			WHERE 
+				dl.link_doctype=%s AND
+				dl.link_name=%s AND
+				dl.parenttype = "Contact"
+			ORDER BY is_billing_contact DESC, is_primary_contact DESC
+		""", (doctype, name))
```
Comment (deepeshgarg007) [inline erpnext/accounts/party.py]: @ronelvcabrera This should be party_type and not "Sales Invoice" which is in this case. Contact will linked to customers and not Sales Invoices. Please update

**Review by deepeshgarg007 — APPROVED** [review]

**Pull request merged**